### PR TITLE
Feat: view current users posts

### LINKED ISSUE #4: Basic Authentication
As the Rare product owner, I would like all users to be authenticated in order to perform any activity in the system so that the system will be able to record which user created post, comment, etc... and EVENTUALLY restrict access to certain features based on user and user type permissions.

**Given** an unauthenticated user is in the Rare application  
**When** they click any link  
**Then** they should be prompted to login using their email address  

**Given** an unauthenticated user is viewing the login form  
**When** they enter an email address that matches an existing User Profile  
**Then** they should be authenticated into the system  
**And** they should be directed to the application home page  

**Given** an unauthenticated user is viewing the login form  
**When** they enter an email address that does not match an existing User Profile  
**Then** an error message should be displayed  
**And** the user should be given another change to enter a valid email address  

> **NOTE:** For the time being it is acceptable to treat all users as `admin` users. There is a future story about enforcing user permissions.

### PR DESCRIPTION
**Pull request by notdanross**
# Description

Added provider and context for posts. 
Added path for getting user posts.
Rendered a list of user posts.

Fixes #4 (server project issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# Testing Instructions

git fetch --all
git checkout dr-viewCurrentUsersPosts
Checkout to corresponding SERVER branch
Navigate to "my posts"
-User MUST have posts inserted into database.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings or errors
- [ ] I have added test instructions that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by daprieu — APPROVED** [review]

**Review by hrlnsnchz — APPROVED** [review]
User posts are listed under the 'My Posts' Section.

File: src/components/posts/PostList.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    useEffect(() => {
+        console.log(posts)
+    }, [posts])
+
+    return (
+        <>
+            <div>
+                {posts.map(post =>
+                    <div className="post_card">
```
Comment (CheoR) [inline src/components/posts/PostList.js]: Warning suggest adding a key.

**Review by CheoR — CHANGES_REQUESTED** [review]
User posts show up as expected.

```bash
Warning: Each child in a list should have a unique "key" prop.
```

File: src/components/posts/PostList.js (hunk 1 above)
Comment (notdanross) [reply src/components/posts/PostList.js]: Just pushed up this change. Thanks for catching this!

**Pull request merged**